src/Engine/ProtoCore/Lang/CallSite.cs

```diff
@@ -1528,6 +1528,27 @@ private FunctionGroup FirstFunctionGroupInInheritanceChain(RuntimeCore runtimeCo
             return ret;
         }
 
+        //Pre-initialize array for repeated calls.  The StackValue is inserted vs making a new array for every call.
+        private static List<StackValue> disposeArguments = new List<StackValue>() { StackValue.Null };
+        //Cache final function endpoint for repeated calls;
+        private FunctionEndPoint finalFep;
+
+        public StackValue DispatchDispose(
+            StackValue stackValue, RuntimeCore runtimeCore)
+        {
+            if (finalFep == null)
+            {
+                var funcGroup = FirstFunctionGroupInInheritanceChain(runtimeCore, classScope);
+                finalFep = funcGroup.FunctionEndPoints[0];
```

Comment:
Why is the Dispose method always the first fep?

Reply:
Has this been tested with DesignScript objects? Is the Dispose method defined on a DS class called after these changes?

Reply:
I pulled some test back out of the past and it did fail.  Not sure how much this case needs to be covered but I updated the PR to handle legacy DS object dispose with the original callr